### ISSUE TITLE
Backend code calling Nim error

### ISSUE THREAD

**Issue by EchoPouet** [nim-lang/Nim#15277]
Hello,

In the documentation that talk about backends (https://nim-lang.org/docs/backends.html) there is the section _Nim invocation example from C_.

I tried to reproduce the example with the **staticLib** option but there are some errors.

I used exactly the same code and I ran this command like the doc.
```bash
$ nim c --app:staticLib --noMain --header fib.nim 
Hint: used config file '/home/bidule/.choosenim/toolchains/nim-1.2.4/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: fib [Processing]
Hint:  [Link]
Hint: 14208 LOC; 0.271 sec; 16.031MiB peakmem; Debug build; proj: /mnt/c/Developpement/test/fib.nim; out: /mnt/c/Developpement/test/fib [SuccessX]
```
The command didn't produce the file **libfib.nim.a**, I did't find it. But I had a **fib** file next to **fib.nim**.

So, I tried with the **fib** file to compile **maths.c** with the following errors.

```bash
$ gcc -o m -I$HOME/.cache/nim/fib_d/ -I$HOME/.choosenim/toolchains/nim-1.2.4/lib/ ./fib maths.c
/usr/bin/ld: /tmp/ccvX75Gg.o: in function `main':
maths.c:(.text+0xd): undefined reference to `NimMain'
/usr/bin/ld: maths.c:(.text+0x20): undefined reference to `fib'
/usr/bin/ld: m: hidden symbol `fib' isn't defined
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

Nim and GCC information:
```bash
$ nim -v
Nim Compiler Version 1.2.6 [Linux: amd64]
$ gcc --version
gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
```

Have you any idea ?

**Comment by jyapayne** [comment]
@ArMour85 I tried to get it working and came up with the following. Run

```bash
nim c --app:staticLib --noMain --header:fib.h fib.nim
```

(linking is needed to produce the static binary `libfib.a`). You should now have `libfib.a` in your `fib.nim` directory.

Then, to compile and link with gcc:

```bash
gcc -o maths -I$HOME/.cache/nim/fib_d -I$HOME/Nim/lib /path/to/libfib.a maths.c
```

Which should produce a `maths` binary.

**Comment by EchoPouet** [comment]
The problem is resolved with Nim version 1.4.0.

Thanks.
